src/PoolDataClient.ts

```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import { Coin, RawPool, PoolData } from "./types";
+
+class PoolDataClient {
+  private poolData: PoolData | null = null;
+  private lastUpdated: number = 0;
+  private expiry = 10000; // 10 seconds
+  private URL = "";
+
+  constructor(dataURL: string) {
+    this.URL = dataURL;
+  }
+
+  async getPoolData(): Promise<PoolData> {
```

Comment:
can we change it so that we pull data from sentio? can find more details at https://app.sentio.xyz/thala-labs/thala-router/data-explorer/sql

```
curl -L -X POST 'https://app.sentio.xyz/api/v1/analytics/thala-labs/thala-router/sql/execute' \
     -H 'api-key: <API_KEY>' \
     -H 'Content-Type: application/json' \
     --data-raw '{
  "sqlQuery": {
    "sql": "-- AI Generated Result:\nSELECT  block_number ,  pool_name ,  contract , asset0, asset1, asset2, asset3,  balance0 ,  balance1 , balance2 , balance3 \nFROM liquidity_change\nWHERE (pool_id, block_number, log_index) IN (\n    SELECT pool_id, block_number, max(log_index)\n    FROM liquidity_change\n    WHERE (pool_id, block_number) IN (\n        SELECT pool_id, max(block_number)\n        FROM liquidity_change\n        GROUP BY pool_id\n    )\n    GROUP BY pool_id, block_number\n);\n",
    "size": 100
  }
}'
```

Reply:
If this is to be a public SDK, we shouldn't leak API key or let users configure their own API key. Still need to host an API to proxy call to sentio

Reply:
ok probably use an endpoint in thala-dapp then